pkg/prometheus/statefulset.go

```diff
@@ -750,7 +750,9 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *operator.Config, shard in
 		}
 
 		if p.Spec.Thanos.ObjectStorageConfig != nil || p.Spec.Thanos.ObjectStorageConfigFile != nil {
-			if p.Spec.Thanos.ObjectStorageConfig != nil {
+			if p.Spec.Thanos.ObjectStorageConfigFile != nil {
+				container.Args = append(container.Args, "--objstore.config-file="+*p.Spec.Thanos.ObjectStorageConfigFile)
+			} else {
```

Comment:
this feels like a duplication of logic that is already present in Thanos itself: https://github.com/thanos-io/thanos/blob/dd670a3e8cd2d99bd86169bdaf6bf304ab5ea2f9/pkg/extflag/pathorcontent.go#L50-L73

As far as i understand we don't really need the if-else switch.

Reply:
Ok.

Reply:
From what I can tell, Thanos won't start if both flags are defined and not empty:
https://github.com/thanos-io/thanos/blob/dd670a3e8cd2d99bd86169bdaf6bf304ab5ea2f9/pkg/extflag/pathorcontent.go#L53-L55

But the [help text](https://github.com/thanos-io/thanos/blob/dd670a3e8cd2d99bd86169bdaf6bf304ab5ea2f9/pkg/extflag/pathorcontent.go#L36) for the `*-file` flag is a bit misleading: it says `Alternative to '...' flag (lower priority). ...`. It should either say that both flags are exclusive or implement the priority logic.

Reply:
I think we can add if-else to keep Semantic correctness.